### PR TITLE
Integration test AEPTestUtils migration

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR updates the integration test classes: 
1. `ConfigOverridesIntegrationTests.kt`
2. `UpstreamIntegrationTests.kt`

to use the AEPTestUtils library, with the following changes:
1. Replace local test helper classes with AEPTestUtils versions
4. Update usages of JSON comparison APIs to reflect:
    1. Automatic conversion of common types into JSONObject/JSONArray - removing manual conversion
    2. Migrating usage of legacy APIs `typeMatchPaths` and `exactMatchPaths`
1. Code style updates
    2. JUnit `assertEquals` is imported directly
    3. `assertExactMatch` and `assertTypeMatch` from JSON comparison is imported directly
    4. Remove `.trimIndent()` from JSON strings
    5. Remove `JSON` suffix from var names
    6. Use `"STRING_TYPE"` convention in expected JSON strings to make value type match intent clearer

Specific changes:
1. `testSendEvent_withPriorState_receivesExpectedStateStoreEventHandle`: Added event expectation to test setup to properly await event processing before beginning test logic

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt [hunk 1]
```diff
@@ -337,15 +341,16 @@ class UpstreamIntegrationTests {
...
+            """
 
             // Unsafe access used since testSendEvent_receivesExpectedEventHandles guarantees existence
             val locationHintResult = TestSetupHelper.getEdgeEventHandles(expectedHandleType = TestConstants.EventSource.LOCATION_HINT_RESULT)
                 .first()
-            JSONAsserts.assertTypeMatch(
-                expected = JSONObject(expectedLocationHintJSON),
-                actual = JSONObject(locationHintResult.eventData),
-                exactMatchPaths = listOf("payload[*].scope", "payload[*].hint")
+            assertTypeMatch(
```
Comment (cacheung) [inline code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt:349]: As discussed, we can do assertExactMatch here and typeMatch the "payload[*].ttlSeconds".

File: code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt [hunk 2]
```diff
@@ -279,26 +282,27 @@ class UpstreamIntegrationTests {
...
 
         // Unsafe access used since testSendEvent_receivesExpectedEventHandles guarantees existence
         val locationHintResult = TestSetupHelper.getEdgeEventHandles(expectedHandleType = TestConstants.EventSource.LOCATION_HINT_RESULT)
             .first()
 
-        JSONAsserts.assertTypeMatch(
-            expected = JSONObject(expectedLocationHintJSON),
-            actual = JSONObject(locationHintResult.eventData),
-            exactMatchPaths = listOf("payload[*].scope")
+        assertTypeMatch(
```
Comment (cacheung) [inline code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt:301]: As discussed, we can do assertExactMatch here and typeMatch the "payload[*].ttlSeconds".

Comment (cacheung) [reply code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt:301]: I am fine with this.

**Review by cacheung — APPROVED** [review]

File: code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt [hunk 3]
```diff
@@ -372,37 +377,32 @@ class UpstreamIntegrationTests {
...
           ]
         }
-        """.trimIndent()
+        """
 
         // Unsafe access used since testSendEvent_receivesExpectedEventHandles guarantees existence
         val stateStoreEvent = TestSetupHelper.getEdgeEventHandles(expectedHandleType = TestConstants.EventSource.STATE_STORE)
             .last()
 
         // Exact match used here to strictly validate `payload` array element count == 2
```
Comment (addb) [inline code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt:401]: nit: update comment

File: code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt [hunk 4]
```diff
@@ -745,9 +745,9 @@ class UpstreamIntegrationTests {
         Edge.sendEvent(experienceEvent) {}
 
         // Verify
-        // Network response assertions
         realNetworkService.assertAllNetworkRequestExpectations()
-        val matchingResponses = realNetworkService.getResponsesFor(networkRequest = invalidNetworkRequest)
+
+        val matchingResponses = realNetworkService.getResponsesFor(invalidNetworkRequest)
 
         Assert.assertEquals(1, matchingResponses?.size)
```
Comment (addb) [inline code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/UpstreamIntegrationTests.kt:752]: Follow similar usages everywhere. 

Import all the required functions like the configOverrideIntegrationTests and update the usages.
```
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
```

**Review by addb — APPROVED** [review]

**Pull request merged**